In [1]:
import cv2
import numpy as np

In [2]:
video_path='data/baby_1.mp4'
cap=cv2.VideoCapture(video_path)

if not cap.isOpened():
    exit()
    
tracker=cv2.TrackerCSRT_create() # ROI를 설정 , CSRT를 사용허겠다고 선언

#img 변수에 첫번째 프레임을 읽어온다.
ret, img = cap.read()# 화면 이미지는 imgresd를 쓴다.


cv2.namedWindow('Select Window')
cv2.imshow('Select Window', img)

rect=cv2.selectROI('Select Window', img, fromCenter=False, showCrosshair= True)
cv2.destroyWindow('Select Window')

#초기화한다.
# img: 처리를 위한 영상 , rect: 영상에서 추척하고 싶은 대상
tracker.init(img,rect)

while True:
    ret,img=cap.read()
    
    if not ret:
        exit()
        
    # success : 영상에서 트래킹의 성공여부 , box: 대상에 대한 정보
    success, box = tracker.update(img)

    left, top, w, h= [int(v) for v in box]

    cv2.rectangle(img, pt1=(left, top), pt2=(left+w, top+h), color=(255,255,255), thickness=3)

    cv2.imshow('img', img)

    if cv2.waitKey(1) ==ord('q'): # 1초기다렸다가 한프레임씩 변수에 넣고 찍는다.
        break

error: OpenCV(4.6.0) D:\a\opencv-python\opencv-python\opencv\modules\imgproc\src\color.cpp:182: error: (-215:Assertion failed) !_src.empty() in function 'cv::cvtColor'


# www.opencv.org 에서 Documentation에서  4.6 다운로드 받아서 함수사용방법이나 설명을 볼수있다.